### PR TITLE
feat: add pagination to GET /nodes endpoint

### LINKED ISSUE #43: Add pagination to GET /nodes endpoint

### PR DESCRIPTION
**Pull request by maxlivinci**
resolves #43

Note:

`from and size` won't work properly if the query result is more than 10,000 items. (In our case when `number_of_results` >= 10,000)

image

[Reference](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html)

### CODE REVIEW

File: services/index/internal/domain/query/es_query_dao.go [hunk 1]
```diff
@@ -30,5 +34,9 @@ func (q *EsQuery) Build() elastic.Query {
 
 	query.Must(subQueries...).Filter(filters...)
 
-	return query
+	return &elastic.Query{
+		Query: query,
+		From:  math.Max(0, q.PageSize*(q.Page-1)),
+		Size:  math.Min(500, math.Max(1, q.PageSize)),
```
Comment (maxlivinci) [inline services/index/internal/domain/query/es_query_dao.go:40]: From:` 0 to any`
Size: `1 to 500`

File: services/index/internal/domain/query/es_query_dto.go [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
+	Lat   *float64 `form:"lat"`
+	Lon   *float64 `form:"lon"`
 	Range *string  `form:"range"`
 
 	Locality *string `form:"locality"`
 	Region   *string `form:"region"`
 	Country  *string `form:"country"`
+
+	Page     float64 `form:"page,default=0"`
+	PageSize float64 `form:"page_size,default=30"`
```
Comment (maxlivinci) [inline services/index/internal/domain/query/es_query_dto.go:16]: Page default 0
PageSize default 30

**Comment by geoffturk** [comment]
There is an error in the counting of `total_pages`. See the failing test in the test suite.

image


image

**Review by geoffturk — CHANGES_REQUESTED** [review]
See above.

**Comment by geoffturk** [comment]
WTF?

https://play.golang.org/p/S42RViSVCBs

**Review by geoffturk — APPROVED** [review]
LGTM 🚀

File: common/pagination/pagination.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ func Size(pageSize int64) int64 {
 }
 
 func TotalPages(numberOfResults int64, pageSize int64) int64 {
-	pages := math.Ceil(float64(numberOfResults / pageSize))
+	pages := math.Ceil(float64(numberOfResults) / float64(pageSize))
```
Comment (maxlivinci) [inline common/pagination/pagination.go:14]: Good catch

**Pull request merged**